datafusion/sql/src/statement.rs

```diff
@@ -670,19 +670,20 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
 
         // Overwrite with assignment expressions
         let mut planner_context = PlannerContext::new();
-        let assign_map = assignments
+        let mut assign_map = assignments
             .iter()
             .map(|assign| {
                 let col_name: &Ident = assign
                     .id
                     .iter()
                     .last()
                     .ok_or(DataFusionError::Plan("Empty column id".to_string()))?;
+                // Validate that the assignment target column exists
+                table_schema.field_with_unqualified_name(&col_name.value)?;
```

Comment:
The logic change looks good to me. I defer to @andygrove and @alamb on adding the `rstest` dependency. Personally I think it's worth it as it appears to reduce test boilerplate and I'm not too worried about `dev-dependencies`.

Reply:
`datafusion-core` already uses rstest https://github.com/apache/arrow-datafusion/blob/be4586858306ebbc47848243fe488dd4316e3f2f/datafusion/core/Cargo.toml#L117 so I don't view this as a new dependency